bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/form/CheckBoxImpl.java

```diff
@@ -49,9 +49,23 @@ public class CheckBoxImpl extends AbstractOptionsFieldImpl implements CheckBox {
     protected String orientationJcr;
     private Orientation orientation;
 
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    private String checkedValue;
```

Comment:
Checked and unchecked value can be even boolean right ? Why is this always a string ?

Reply:
Saving the values as string in crx. But while getting from sling model, the type-cast will happen to the data type